test/framework/kubetest/run.go

```diff
@@ -143,27 +137,51 @@ func Run(ctx context.Context, input RunInput) error {
 	if input.ConformanceImage == "" {
 		input.ConformanceImage = versionToConformanceImage(input.KubernetesVersion)
 	}
-	kubeConfigVolumeMount := volumeArg(tmpKubeConfigPath, "/tmp/kubeconfig")
-	outputVolumeMount := volumeArg(reportDir, "/output")
-	viperVolumeMount := volumeArg(tmpConfigFilePath, "/tmp/viper-config.yaml")
+	volumeMounts := map[string]string{
+		tmpKubeConfigPath: "/tmp/kubeconfig",
+		reportDir:         "/output",
+		tmpConfigFilePath: "/tmp/viper-config.yaml",
+	}
 	user, err := user.Current()
 	if err != nil {
 		return errors.Wrap(err, "unable to determine current user")
 	}
-	userArg := user.Uid + ":" + user.Gid
-	networkArg := "--network=kind"
-	e2eCmd := exec.Command("docker", "run", "--user", userArg, kubeConfigVolumeMount, outputVolumeMount, viperVolumeMount, "-t", networkArg)
-	if len(testRepoListVolumeArgs) > 0 {
-		e2eCmd.Args = append(e2eCmd.Args, testRepoListVolumeArgs...)
+	env := map[string]string{}
+
+	if input.KubeTestRepoListPath != "" {
+		tmpKubeTestRepoListPath := path.Join(kubetestConfigDir, "repo_list.yaml")
+		if err := copyFile(input.KubeTestRepoListPath, tmpKubeTestRepoListPath); err != nil {
+			return err
+		}
+		dest := "/tmp/repo_list.yaml"
+		env["KUBE_TEST_REPO_LIST"] = dest
+		volumeMounts[tmpKubeTestRepoListPath] = dest
 	}
-	e2eCmd.Args = append(e2eCmd.Args, input.ConformanceImage)
-	e2eCmd.Args = append(e2eCmd.Args, "/usr/local/bin/ginkgo")
-	e2eCmd.Args = append(e2eCmd.Args, ginkgoArgs...)
-	e2eCmd.Args = append(e2eCmd.Args, "/usr/local/bin/e2e.test")
-	e2eCmd.Args = append(e2eCmd.Args, "--")
-	e2eCmd.Args = append(e2eCmd.Args, e2eArgs...)
-	e2eCmd = framework.CompleteCommand(e2eCmd, "Running e2e test", false)
-	if err := e2eCmd.Run(); err != nil {
+
+	// Formulate our command arguments
+	args := []string{"/usr/local/bin/ginkgo"}
+	args = append(args, ginkgoArgs...)
+	args = append(args, "/usr/local/bin/e2e.test")
+	args = append(args, "--")
+	args = append(args, e2eArgs...)
+
+	// Get our current working directory. Just for information, so we don't need
+	// to worry about errors at this point.
+	cwd, _ := os.Getwd()
+	Byf("Running e2e test: dir=%s, command=%q", cwd, args)
+
+	err = dockerexec.RunContainer(
```

Comment:
Maybe we should use the "input struct" pattern as in large parts of the test framework. I think the idea is that it's easier readable because you directly see the "parameter" names

Reply:
Done